OpenRA.Game/FieldLoader.cs

```diff
@@ -467,202 +479,307 @@ public static object GetValue(string fieldName, Type fieldType, MiniYaml yaml, M
+					return InvalidValueAction(value, fieldType, fieldName);
+
+				var ints = new int2[parts.Length / 2];
+				for (var i = 0; i < ints.Length; i++)
+					ints[i] = new int2(Exts.ParseIntegerInvariant(parts[2 * i]), Exts.ParseIntegerInvariant(parts[2 * i + 1]));
+
+				return ints;
+			}
+
+			return InvalidValueAction(value, fieldType, fieldName);
+		}
+
+		static object GetValue(string fieldName, Type fieldType, MiniYaml yaml, MemberInfo field)
+		{
+			var value = yaml.Value;
+			if (value != null) value = value.Trim();
+
+			switch (fieldType.Name)
+			{
+				case "Int16":
+				case "Int32":
+				case "Int64":
+				case "UInt16":
+				case "UInt32":
+				case "UInt64":
+				case "Decimal":
+				case "Double":
+				case "SByte":
+				case "Single":
+					return ParseNumerics(fieldType, value, fieldName);
+				case "String":
+					return ParseString(fieldName, value, field);
+				case "Color":
+					return ParseColor(fieldName, value, fieldType);
+				case "Color[]":
+					return ParseColors(fieldName, value, fieldType);
+				case "HSLColor":
+					return ParseHSLColor(fieldName, value, fieldType);
+				case "Hotkey":
+					return ParseHotkey(fieldName, value, fieldType);
+				case "HotkeyReference":
+					return ParseHotkeyReference(fieldName, value, fieldType);
+				case "WDist":
+					return ParseWDistance(fieldName, value, fieldType);
+				case "WVec":
+					return ParseWVector(fieldName, value, fieldType);
+				case "WVec[]":
+					return ParseWVectors(fieldName, value, fieldType);
+				case "WPos":
+					return ParseWPosition(fieldName, value, fieldType);
+				case "WAngle":
+					return ParseWAngle(fieldName, value, fieldType);
+				case "WRot":
+					return ParseWRot(fieldName, value, fieldType);
+				case "CPos":
+					return ParseCPosition(fieldName, value, fieldType);
+				case "CVec":
+					return ParseCVector(fieldName, value, fieldType);
+				case "CVec[]":
+					return ParseCVectors(fieldName, value, fieldType);
+				case "BooleanExpression":
+					return ParseBoolExpression(fieldName, value, fieldType);
+				case "IntegerExpression":
+					return ParseIntegerExpression(fieldName, value, fieldType);
+				case "ImageFormat":
+					return ParseImageFormat(fieldName, value, fieldType);
+				case "Boolean":
+					return ParseBool(fieldName, value, fieldType);
+				case "HashSet`1":
+					return ParseHashSet(fieldName, value, fieldType, field);
+				case "Dictionary`2":
+					return ParseDict(fieldName, value, fieldType, yaml, field);
+				case "Size":
+					return ParseSize(fieldName, value, fieldType);
+				case "int2":
+					return ParsePairInt(fieldName, value, fieldType);
+				case "float2":
+					return ParsePairFloat(fieldName, value, fieldType);
+				case "float3":
+					return ParseTripletFloat(fieldName, value, fieldType);
+				case "Rectangle":
+					return ParseRectangle(fieldName, value, fieldType);
+				case "DateTime":
+					return ParseDateTime(fieldName, value, fieldType);
+				case "BitSet`1":
```

Comment:
Typo.

Reply:
Are you referring to lines 747, 749 and 763? 

If so, then those are actual names of the relevant types.

Reply:
Well values of their Type.Name property.

Reply:
You mean that the ``    `NUMBER`` suffix is part of the type?

Reply:
Yeah. It thought it looked weird, but that's actually the value of the relevant `.Name` property.